lib/fxpay/index.js

```diff
@@ -22,15 +22,29 @@
 
 
   exports.init = function _init(opt) {
+    utils.logDeprecation(
+      'fxpay.init() is no longer supported; use ' +
+      'fxpay.getProducts()...product.validateReceipt() instead', '0.0.15');
+
     settings.initialize(opt);
 
-    settings.adapter.init(function(err) {
-      if (err) {
-        return settings.onerror(err);
-      }
+    fxpay.getProducts()
+      .then(function(products) {
+        products.forEach(function(product) {
+
+          if (product.hasReceipt()) {
+            product.validateReceipt().then(function(productInfo) {
+              settings.onrestore(null, productInfo);
+            }).catch(function(error) {
+              settings.onrestore(error, error.productInfo);
+            });
+          }
+
+        });
+      })
+      .then(settings.oninit)
```

Comment:
I think this was always the case but I think I'm correct in saying oninit is fired even if the receipt checking is still underway, that's probably OK but would be worth making that clearer in the docs https://developer.mozilla.org/en-US/Marketplace/Monetization/In-app_payments_section/fxPay_iap#Initialization